pkg/kv/kvserver/scheduler.go

```diff
@@ -297,7 +303,7 @@ func (s *raftScheduler) worker(ctx context.Context) {
 				state.flags |= stateRaftReady
 			}
 		}
-		if state.flags&stateRaftTick != 0 {
+		for tick, t := state.flags&stateRaftTick != 0, state.ticks; tick && t > 0; t-- {
```

Comment:
nit: may want to just have an if wrapping the for-loop here. While I appreciate the terseness, it's a bit hard to parse.